client/client.go

```diff
@@ -95,6 +96,42 @@ func (c *Client) BuildRequest(method, path string, body interface{}) (*http.Requ
 	return req, nil
 }
 
+// Marshalls the request body and returns the resulting byte slice
+// This is split out of the BuildRequestBody method so as to allow
+// the developer to print a debug string of the request body if they
+// should so choose.
+func (c *Client) MarshallRequestBody(body interface{}) ([]byte, error) {
+	// Verify interface isnt' nil
+	if body == nil {
+		return nil, nil
```

Comment:
should this not return an error?

Reply:
Aye, because some PaaS resources have all of their configuration as HTTP path parameters, with a nil body. If we don't pass a body into the method, don't need to error out, as it might be desired.